src/pal/src/map/virtual.cpp

```diff
@@ -885,10 +889,8 @@ static LPVOID VIRTUALReserveMemory(
     // First, figure out where we're trying to reserve the memory and
     // how much we need. On most systems, requests to mmap must be
     // page-aligned and at multiples of the page size.
-    StartBoundary = (UINT_PTR)lpAddress & ~BOUNDARY_64K;
```

Comment:
I think this needs to stay to be compatible with Windows. I am not sure, but I think there was some code in the VM or JIT that relies on that.